### DISCUSSION TITLE
Grass Desktop Client

### DISCUSSION THREAD

**Discussion by copystring** [engageub/InternetIncome#158] [General]
Hi!

Did you see, there's a grass desktop client not relying on browser? Do you think it's possible to switch your script to this?
See https://app.getgrass.io/dashboard/store/item/desktop

**Comment by engageub** [comment]
Hello,
Yes, it is possible if linux app is containarized.
Either we wait for them to release the docker image or look for unofficial
version which uses the node.

Thank You

**Comment by copystring** [comment]
What about this one? https://github.com/Trangoul1988/docker-grass-desktop

**Reply by engageub** [reply]
I attempted to use that, but I am currently unable to log in to my account.
Additionally, I could not locate the device ID while online during my attempts.
I also tried to reset the password but did not receive any email confirmation.
Have you had the opportunity to use the Docker image and observed the device online in your dashboard?

Thank you

**Reply by copystring** [reply]
I tried this out.
Works perfectly fine.

![{CB591D65-E41E-4C56-BEDF-BAF55FDB13E6}](https://github.com/user-attachments/assets/c6a34059-f477-4132-819e-155c454de956)

**Reply by engageub** [reply]
Hello,
Thank you for the confirmation. The changes are live now.
Please download the branch again to use the latest code. 

Thank you

**Comment by copystring** [comment]
I updated about 2 hours ago. Looking good so far.

**Comment by copystring** [comment]
There is a problem with this. After a few hours, the web interface shows the desktop client as offline. Restarting the container helps. Maybe implement a restart similar to browser for ebesucher?

**Reply by engageub** [reply]
Hello,
We will have to find out what is causing the issue.
Ebesucher and Adnade are being restarted as they progressively consume memory while running. To avoid high memory usage, they are being restarted hourly after clearing the browser data.
This is a desktop application and should reconnect automatically if there are any connectivity issues.
Does this happen with direct connection also or only while using proxies?
Would you be able to run direct connection with VNC open. If there is connectivity issue, you can look into VNC container to see if there are any issues with Application. Also, please enable logs, so that we can have a look into docker container logs. 
If the linux container, goes to sleep mode after sometime, we will have to contact the author of docker image to remove such restrictions. 


Thank you

**Comment by engageub** [comment]
I had a look into the discord forum and other users have issues related to Grass Desktop version being disconnected.
The user was asked to wait for the update. Please find the screenshot below.

![image](https://github.com/user-attachments/assets/927324a9-e6e1-48dd-9285-cc9b5cd04dca)

**Reply by engageub** [reply]
Hello,
I began using the GRASS Desktop Client today, and I noticed that the latest version automatically reconnects after disconnection. However, it appears that the Docker image needs to be updated to align with this version, and I have raised an issue regarding this matter.

If you are familiar with utilizing the Dockerfile to create an image, you can refer to the code provided in the issue. Alternatively, you may choose to wait for the user to release the updated Docker image.

https://github.com/Trangoul1988/docker-grass-desktop/issues/1


Thank you

**Reply by Raffy142** [reply]
sorry i know it's not relevant to the discussion, but are you able to get the payment from grass? have you ever received money? how do you manage to get it if you do?

**Reply by engageub** [reply]
Hello,
Yes, I received the recent grass airdrop. 
You need some SOL tokens for fee. 
Make sure to increase the slippage to 1% if the default slippage does not work. 
Please claim your tokens here.

https://www.grassfoundation.io/claim

Thank you

**Reply by Raffy142** [reply]
sorry, how often do they do an airdrop?

**Comment by engageub** [comment]
Hello,
If you are using phantom wallet, you may follow this tutorial.

https://m.youtube.com/watch?v=YekZQHwYhIE&pp=ygUTUGhhbnRvbSB3YWxsZXQgc3dhcA%3D%3D

If you have queries related to airdrop, please join their discord channel
to get more details.

Thank You

**Comment by Raffy142** [comment]
I’m getting this when i run getgrass desktop
 With linux, i’m using proxy too. U have some solution?


/usr/bin/rm: cannot remove '/tmp/.X1-lock': No such file or directory
/usr/bin/rm: cannot remove '/tmp/.X11-unix': No such file or directory
VNC_PASSWORD environment variable is not set. Using a random password. You
will not be able to access the VNC server.
/usr/bin/xauth:  file /home/grass/.Xauthority does not exist

Desktop 'TurboVNC: 9f8664b1cfd5:1 ()' started on display 9f8664b1cfd5:1

Starting applications specified in /opt/TurboVNC/bin/xstartup.turbovnc
Log file is /home/grass/.vnc/9f8664b1cfd5:1.log

MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen

(grass:87): dbind-WARNING **: 14:09:51.410: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen
Could not determine the accessibility bus address
/usr/bin/grass

**Comment by engageub** [comment]
That should be fine. But the docker image has to be
 updated. Please find the issue below. If the author does not update the
image soon, will have to override the code to make it work.

 https://github.com/Trangoul1988/docker-grass-desktop/issues/1

Thank you

**Reply by engageub** [reply]
Hello,
Please find the list of CEX and DEX where grass is being traded.

https://coinmarketcap.com/currencies/grass/#Markets

Thank you

**Comment by hieunguyen8794** [comment]
get error when start grass
`rm: cannot remove '/tmp/.X1-lock': No such file or directory
rm: cannot remove '/tmp/.X11-unix': No such file or directory
VNC_PASSWORD environment variable is not set. Using a random password. You
will not be able to access the VNC server.
xauth:  file /home/grass/.Xauthority does not exist

Desktop 'TurboVNC: 6b82df2ca9fc:1 ()' started on display 6b82df2ca9fc:1

Starting applications specified in /opt/TurboVNC/bin/xstartup.turbovnc
Log file is /home/grass/.vnc/6b82df2ca9fc:1.log

MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen

(grass:35): dbind-WARNING **: 06:16:43.616: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen

(grass:35): libayatana-appindicator-WARNING **: 06:16:43.838: Unable to get the session bus: Failed to close file descriptor for child process (Operation not permitted)

(grass:35): LIBDBUSMENU-GLIB-WARNING **: 06:16:43.838: Unable to get session bus: Failed to close file descriptor for child process (Operation not permitted)
/usr/bin/grass`

**Comment by engageub** [comment]
Hello,
These errors are common but the problem is with auto-login which I asked the author to update. 

https://github.com/Trangoul1988/docker-grass-desktop/issues/1

I have a working solution to override the file but wanted the author to update that in docker image itself. 

Thank you

**Comment by engageub** [comment]
A temporary update has been made in test branch to resolve grass auto-login.

**Reply by hieunguyen8794** [reply]
it's still error, i use proxy
`/InternetIncome-test# docker logs grass1e6e4eb7ab7dab6b13ee4d853848d03a18
rm: cannot remove '/tmp/.X1-lock': No such file or directory
rm: cannot remove '/tmp/.X11-unix': No such file or directory
VNC_PASSWORD environment variable is not set. Using a random password. You
will not be able to access the VNC server.
xauth:  file /home/grass/.Xauthority does not exist

Desktop 'TurboVNC: 2f13641b4c41:1 ()' started on display 2f13641b4c41:1

Starting applications specified in /opt/TurboVNC/bin/xstartup.turbovnc
Log file is /home/grass/.vnc/2f13641b4c41:1.log

MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen

(grass:36): dbind-WARNING **: 09:37:51.410: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen

(grass:36): libayatana-appindicator-WARNING **: 09:37:51.724: Unable to get the session bus: Failed to close file descriptor for child process (Operation not permitted)

(grass:36): LIBDBUSMENU-GLIB-WARNING **: 09:37:51.725: Unable to get session bus: Failed to close file descriptor for child process (Operation not permitted)
/usr/bin/grass
root@ubuntu-s-4vcpu-8gb-nyc3-02:~/InternetIncome-test# ls
README.md              containernames.txt  earnappdata             grass_start.sh     properties.conf  restart.sh        vpns.txt
banner.jpg             containers.txt      firefoxprofiledata.zip  internetIncome.sh  proxies.txt      test.zip
chromeprofiledata.zip  earnapp.txt         generateDeviceIds.sh    multi_ips.txt      resolv.conf      updateProxies.sh
root@ubuntu-s-4vcpu-8gb-nyc3-02:~/InternetIncome-test# cat grass_start.sh 
#!/bin/bash

# Enable job control
set -m

# These files could be left-over if the container is not shut down cleanly. We just remove it since we should
# only be here during container startup.
rm /tmp/.X1-lock
rm -r /tmp/.X11-unix

# Set up the VNC password
if [ -z "$VNC_PASSWORD" ]; then
    echo "VNC_PASSWORD environment variable is not set. Using a random password. You"
    echo "will not be able to access the VNC server."
    VNC_PASSWORD="$(tr -dc '[:alpha:]' < /dev/urandom | fold -w "${1:-8}" | head -n1)"
fi
mkdir ~/.vnc
echo -n "$VNC_PASSWORD" | /opt/TurboVNC/bin/vncpasswd -f > ~/.vnc/passwd
chmod 400 ~/.vnc/passwd
unset VNC_PASSWORD

# TurboVNC by default will fork itself, so no need to do anything here
/opt/TurboVNC/bin/vncserver -rfbauth ~/.vnc/passwd -geometry 1200x800 -rfbport 5900 -wm openbox :1

export DISPLAY=:1

if [ -z "$GRASS_USERNAME" ] || [ -z "$GRASS_PASSWORD" ]; then
    >&2 echo "The GRASS_USERNAME and GRASS_PASSWORD environment variables need to be set"
    >&2 echo "before docker-grass-desktop can start. If you do not already have a username"
    >&2 echo "and password, sign up in a browser at:"
    >&2 echo "https://app.getgrass.io/register/?referralCode=DLZzmgbPgg46WUJ"
    >&2 echo "The container will now exit."
    exit 243
fi

/usr/bin/grass &

if ! [ -f ~/.grass-configured ]; then
    # Wait for the grass window to be available
    while [[ "$(xdotool search --name Grass | wc -l)" -lt 3 ]]; do
        sleep 10
    done

    # Handle grass login
    xdotool search --name Grass | tail -n1 | xargs xdotool windowfocus
    sleep 5
    xdotool key Tab
    xdotool key Return
    sleep 5
    xdotool key Tab
    xdotool key Tab
    xdotool key Tab
    xdotool type "$GRASS_USERNAME"
    xdotool key Tab
    xdotool type "$GRASS_PASSWORD"
    xdotool key Return
    sleep 5
    xdotool key Escape

    touch ~/.grass-configured
fi

fg %/usr/bin/grass`

**Comment by engageub** [comment]
Hello,
Those warning messages are common at the beginning. Please make sure that you see your device online in grass dashboard.


Thank you

**Comment by engageub** [comment]
Hello,
The author has updated the grass docker image with necessary changes.
Please delete and start the script to auto-download the latest docker image if you are using main branch. 


Thank you

**Reply by Raffy142** [reply]
Hi. I’m getting the same problem after update docker image

rm: cannot remove '/tmp/.X1-lock': No such file or directory
rm: cannot remove '/tmp/.X11-unix': No such file or directory
VNC_PASSWORD environment variable is not set. Using a random password. You
will not be able to access the VNC server.
xauth:  file /home/grass/.Xauthority does not exist

Desktop 'TurboVNC: c8bfd3a8fc83:1 ()' started on display c8bfd3a8fc83:1

Starting applications specified in /opt/TurboVNC/bin/xstartup.turbovnc
Log file is /home/grass/.vnc/c8bfd3a8fc83:1.log

MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
glx: failed to create drisw screen

(grass:34): dbind-WARNING **: 00:09:20.907: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen

(grass:34): libayatana-appindicator-WARNING **: 00:09:21.125: Unable to get the session bus: Failed to close file descriptor for child process (Operation not permitted)

(grass:34): LIBDBUSMENU-GLIB-WARNING **: 00:09:21.125: Unable to get session bus: Failed to close file descriptor for child process (Operation not permitted)

**Reply by engageub** [reply]
Hello,
As mentioned earlier, the warning messages are common and is related to docker image build. You will have to contact the author to remove those warning messages in logs. 

Make sure your device is online in grass dashboard.

Please wait for a minute for the device to be online in dashboard.

Thank you

**Comment by hieunguyen8794** [comment]
The dashboard only shows the connection of the first proxy. The other proxies cannot connect. I have deleted the old image and rerun the script with --start, but it still doesn’t work. The earn app still connects to all proxies, grass still only first

**Reply by hieunguyen8794** [reply]
![image](https://github.com/user-attachments/assets/6367b732-919c-4b0b-bcdf-369e3803b363)

**Comment by engageub** [comment]
Hello,
At the bottom of the page, please look for the "Load More" button. If it is visible, click on it; if it is not, that may indicate that all devices have already been loaded.

Note that devices may sometimes appear at the end or after all others have been loaded.

Additionally, ensure that your CPU usage does not reach 100% when starting the container. If it does peak at 100% for the initial container, please set a delay of 1 minute in properties.conf. As this application is a GUI-based program, it may consume more CPU resources than others, particularly during the initial loading phase, which can result in a temporary spike in CPU usage.

Thank you

**Comment by hieunguyen8794** [comment]
![image](https://github.com/user-attachments/assets/e97ffc99-9ebc-40be-961f-7242b50e6732)
I don’t think performance is the issue because there are still plenty of resources. Previously, I successfully connected 10 proxies, all set to connect. But when I updated to 20 proxies, only the first proxy could connect. I also tried clicking “load more,” but all the devices below still couldn’t connect.

**Reply by engageub** [reply]
Hello,
You will have to look into performance when the container starts. 
There is a sudden spike in CPU when it starts. Please open a new tab and monitor the CPU by checking the "IDLE" percentage.
If you are using `top` command please check the percentage beside `id`. That's the amount of CPU % left on your device. 
 Try to give a delay of 1 minute between the containers and then start it.
You will see devices being online one by one.

Thank you

**Comment by engageub** [comment]
I looked into my account and have about 20 devices. 
I don't see any devices being online on my account now.  The earnings are increasing. 
May be the devices limit to be displayed in the dashboard has reached.
Please note that grass dashboard displays the aggregate earnings the next day.
Even if it shows 1000 points or less, the next day it may be 23,000 for the current day.


Thank you

**Reply by engageub** [reply]
If you look at my dashboard for November 3, it is still updating the previous days data.
The update is delayed in grass dashboard. 

![image](https://github.com/user-attachments/assets/493b69b6-cabc-40c4-9142-0f688614544f)
